python/hsfs/feature_group.py

```diff
@@ -1289,6 +1290,9 @@ def __init__(
 
         if self._id:
             # Got from Hopsworks, deserialize features and storage connector
+            self.primary_key = [
+                feat.name for feat in self._features if feat.primary is True
+            ]
```

Comment:
This doesn't work. `self._feature` is empty until the next line.

Reply:
Indeed, if you try to do `fs.get_on_demand_feature_group("name_of_fg")` it fails.